pkg/traefik/traefik.go

```diff
@@ -0,0 +1,310 @@
+package traefik
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/celestiaorg/knuu/pkg/k8s"
+	"github.com/celestiaorg/knuu/pkg/names"
+	"github.com/sirupsen/logrus"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/utils/ptr"
+)
+
+const (
+	traefikServiceName = "traefik"
+	Port               = 80
+	PortSecure         = 443
+	deploymentName     = "traefik-deployment"
+	roleName           = "traefik-role"
+	containerName      = "traefik"
+	image              = "traefik:v3.0"
+	appLabel           = "app"
+	appLabelValue      = "traefik"
+	replicas           = 2
```

Comment:
Do we actually need two replicas?

Reply:
I am not sure, probably not for most tests. However, I think we need to make it scalable somehow so if there is a large scale test it can scale up itself.